src/main/groovy/wooga/gradle/build/unity/tasks/AbstractUnityBuildEngineTask.groovy

```diff
@@ -0,0 +1,161 @@
+package wooga.gradle.build.unity.tasks
+
+
+import org.gradle.api.file.Directory
+import org.gradle.api.file.RegularFile
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Optional
+import wooga.gradle.build.unity.internal.BuildEngineArgs
+import wooga.gradle.secrets.internal.Secrets
+import wooga.gradle.unity.UnityTask
+
+import javax.crypto.spec.SecretKeySpec
+
+
+abstract class AbstractUnityBuildEngineTask extends UnityTask {

```

Comment:
For the base class name, I think AbstractBuildEngineUnityTask would be clearer as it would denote the hierarchy better:
"This is an Unity Task for the Build Engine". 

Perhaps go a bit further and name it BuildEngineUnityTask.

Reply:
Yeah, I do agree that your naming is way better. Also, I never liked "Abstract" on class name as well. I renamed AbstractBuildEngineUnityTask  to BuildEngineUnityTask, and the previous BuildEngineUnityTask to BasicBuildEngineUnityTask